### PR TITLE
[FEATURE] `DataContextVariables` CRUD for `anonymous_usage_statistics`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Continuing pattern defined by GREAT-937 to add accessors for `anonymous_usage_statistics`

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/data_context/test_data_context_variables.py [hunk 1]
```diff
@@ -38,11 +39,11 @@ def data_context_config_dict() -> dict:
             },
         },
         "data_docs_sites": {},
-        "anonymous_usage_statistics": {
-            "enabled": True,
-            "data_context_id": "6a52bdfa-e182-455b-a825-e69f076e67d6",
-            "usage_statistics_url": "https://www.my_usage_stats_url/test",
-        },
+        "anonymous_usage_statistics": AnonymizedUsageStatisticsConfig(
```
Comment (Shinnnyshinshin) [inline tests/data_context/test_data_context_variables.py:42]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]
love it

File: great_expectations/data_context/types/data_context_variables.py [hunk 2]
```diff
@@ -20,7 +21,7 @@ class DataContextVariableSchema(str, enum.Enum):
     DATA_DOCS_SITES = "data_docs_sites"
     NOTEBOOKS = "notebooks"
     CONFIG_VARIABLES_FILE_PATH = "config_variables_file_path"
-    ANONYMIZED_USAGE_STATISTICS = "anonymous_usage_statistics"
+    ANONYMOUS_USAGE_STATISTICS = "anonymous_usage_statistics"
```
Comment (anthonyburdi) [inline great_expectations/data_context/types/data_context_variables.py:24]: Oooh confusing. Good catch.

**Review by anthonyburdi — APPROVED** [review]
LGTM!

**Pull request merged**